resource-service/common/git.go

```diff
@@ -457,6 +457,7 @@ func (g *Git) CheckoutBranch(gitContext common_models.GitContext, branch string)
 
 	err := g.checkoutBranch(gitContext, &git.CheckoutOptions{
 		Branch: b,
+		Force:  true,
```

Comment:
Do we know what changes would be lost if we do a force checkout?

Reply:
Only the resource-service can produce unstaged changes on our local branch.

Reply:
I don't have much insights, but I would expect that a failed operation (e.g., due to ctx cancellation) can trigger unstashed data